client/post-editor/editor-drawer/index.jsx

```diff
@@ -134,14 +135,16 @@ const EditorDrawer = React.createClass( {
 				! this.currentPostTypeSupports( 'post-formats' ) ) {
 			return;
 		}
-
 		return (
-			<PostFormatsData siteId={ this.props.site.ID }>
+			<div>
+				<QueryPostFormats siteId={ this.props.site.ID } />
```

Comment:
We should strive for components to be self-sufficient with query components and `connect` applied as close to the UI requirements as possible. To that point, at the very least it should be the `<EditorPostFormatsAccordion />` component which renders the `<QueryPostFormats />` and retrieves `postFormats` from state. I might go one step further and say that both `<EditorPostFormatsAccordion />` and `<EditorPostFormats />` should separately connect to pull from `postFormats` state so that they're not so tightly coupled.

In the end, I'd love if `<EditorDrawer />` were simply a render function like:

```jsx
function EditorDrawer() {
	return (
		<div className="editor-drawer">
			<EditorDrawerTaxonomies />
			<EditorDrawerFeaturedImage />
			<EditorDrawerPageOptions />
			<EditorDrawerSharing />
			<EditorDrawerPostFormats />
			<EditorDrawerSeo />
			<EditorDrawerMoreOptions />
		</div>
	);
}
```

Where each component is responsible for deciding itself whether to render and how.